### PR TITLE
Bug/update create user validators

### PR DESCRIPTION
**Pull request by Cheephot**

### CODE REVIEW

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 1]
```diff
@@ -8,22 +8,34 @@ namespace LT.DigitalOffice.UserService.Validation.User
...
+        private static Regex SpecialCharactersRegex = new (@"[$&+,:;=?@#|'<>.^*()%!-]");
+        private static Regex SpaceRegex = new (@"^[^@\s]*$");
+        private static Regex EmailRegex = new (@"^[^@\s]+@[^@\s]+\.[^@\s]*$");
+
         public CreateUserRequestValidator()
         {
             RuleFor(user => user.FirstName)
                 .NotEmpty()
-                .Must(x => !NameRegex.IsMatch(x))
+                .WithMessage("The field cannot be empty")
```
Comment (MariyaSin) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:20]: write "The first name cannot be empty", not "field"

**Review by MariyaSin — APPROVED** [review]

File: src/UserService.Validation/User/CreateUserRequestValidator.cs (hunk 1 above)
Comment (Cheephot) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

**Review by BelehovEgor — APPROVED** [review]

File: src/UserService.Validation/Communication/CreateCommunicationRequestValidator.cs [hunk 2]
```diff
@@ -30,7 +30,7 @@ public CreateCommunicationRequestValidator(IUserRepository userRepository)
                 RuleFor(x => x.Value)
                     .Must(v => EmailRegex.IsMatch(v.Trim()))
                     .WithMessage("Incorrect email address."));
-
+
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/UserService.Validation/Communication/CreateCommunicationRequestValidator.cs]: done

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 3]
```diff
@@ -8,22 +8,34 @@ namespace LT.DigitalOffice.UserService.Validation.User
...
-                .Must(x => !NameRegex.IsMatch(x))
+                .WithMessage("The first name cannot be empty")
+                .Must(x => !NumberRegex.IsMatch(x))
                 .WithMessage("First name must not contain numbers")
+                .Must(x => !SpecialCharactersRegex.IsMatch(x))
+                .WithMessage("First name must not contain special characters")
                 .MaximumLength(32)
-                .WithMessage("First name is too long.");
+                .WithMessage("First name is too long.")
+                .Must(x => SpaceRegex.IsMatch(x.Trim()));
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 4]
```diff
@@ -8,22 +8,34 @@ namespace LT.DigitalOffice.UserService.Validation.User
...
-                .Must(x => !NameRegex.IsMatch(x))
+                .WithMessage("The last name cannot be empty")
+                .Must(x => !NumberRegex.IsMatch(x))
                 .WithMessage("Last name must not contain numbers")
+                .Must(x => !SpecialCharactersRegex.IsMatch(x))
+                .WithMessage("Last name must not contain special characters")
                 .MaximumLength(32)
-                .WithMessage("Last name is too long.");
+                .WithMessage("Last name is too long.")
+                .Must(x => SpaceRegex.IsMatch(x.Trim()));
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 5]
```diff
@@ -32,10 +44,13 @@ public CreateUserRequestValidator()
...
                     RuleFor(user => user.MiddleName)
-                        .Must(x => !NameRegex.IsMatch(x))
+                        .Must(x => !NumberRegex.IsMatch(x))
                         .WithMessage("Middle name must not contain numbers")
+                        .Must(x => !SpecialCharactersRegex.IsMatch(x))
+                        .WithMessage("Middle name must not contain special characters")
                         .MaximumLength(32)
-                        .WithMessage("Middle name is too long."));
+                        .WithMessage("Middle name is too long.")
+                        .Must(x => SpaceRegex.IsMatch(x.Trim())));
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 6]
```diff
@@ -57,7 +72,11 @@ public CreateUserRequestValidator()
                 RuleForEach(user => user.Communications)
                     .ChildRules(c =>
                     {
-                        c.RuleFor(uc => uc.Value).NotEmpty();
+                        c.RuleFor(uc => uc.Value)
+                        .NotEmpty()
+                        .WithMessage("The email cannot be empty")
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 7]
```diff
@@ -57,7 +72,11 @@ public CreateUserRequestValidator()
...
+                        .Must(x => EmailRegex.IsMatch(x))
+                        .WithMessage("Incorrect email address.");
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

**Review by ndoriomedov — APPROVED** [review]

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 8]
```diff
@@ -32,17 +47,28 @@ public CreateUserRequestValidator()
...
                 () =>
                     RuleFor(user => user.City)
                         .MaximumLength(32)
-                        .WithMessage("City name is too long."));
+                        .WithMessage("City name is too long.")
+                        .Must(x => !NumberRegex.IsMatch(x))
+                        .WithMessage("City name must not contain numbers.")
+                        .Must(x => !SpecialCharactersRegex.IsMatch(x))
+                        .WithMessage("City name must not contain special characters.")
+                        .MaximumLength(32)
```
Comment (MariyaSin) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:69]: add message

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 9]
```diff
@@ -66,6 +96,15 @@ public CreateUserRequestValidator()
...
 
+            RuleFor(user => user.Password)
+                .NotEmpty()
+                .WithMessage("The password cannot be empty.")
+                .MinimumLength(5)
+                .WithMessage("Password is too short.")
+                .Must(x => SpaceRegex.IsMatch(x))
+                .WithMessage("Password must not contain space.");
+
+
```
Comment (MariyaSin) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:107]: remove empty line

**Review by MariyaSin — APPROVED** [review]

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 10]
```diff
@@ -32,17 +47,28 @@ public CreateUserRequestValidator()
...
-                        .WithMessage("Middle name is too long."));
+                        .WithMessage("Middle name is too long.")
+                        .Must(x => NameRegex.IsMatch(x.Trim()))
+                        .WithMessage("Middle name contains invalid characters."));
 
             When(
                 user => !string.IsNullOrEmpty(user.City),
                 () =>
                     RuleFor(user => user.City)
                         .MaximumLength(32)
```
Comment (Glunky) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:63]: why 32? use constant

**Review by Glunky — APPROVED** [review]

File: src/UserService.Validation/User/CreateUserRequestValidator.cs (hunk 10 above)
Comment (MariyaSin) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs:63]: We usually write value, without variables

**Pull request merged**